File: problems/1389-create-target-array-in-the-given-order/solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] createTargetArray(int[] nums, int[] index) {
        List<Integer> targetList = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            targetList.add(index[i], nums[i]);
        }
        int[] targetArray = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            targetArray[i] = targetList.get(i);
        }
        return targetArray;
    }
}

File: problems/1389-create-target-array-in-the-given-order/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* createTargetArray(int* nums, int numsSize, int* index, int indexSize, int* returnSize) {
    int* target = (int*)malloc(sizeof(int) * numsSize);
    *returnSize = 0;
    for (int i = 0; i < numsSize; i++) {
        int idx = index[i];
        if (idx == *returnSize) {
            target[*returnSize] = nums[i];
            (*returnSize)++;
        } else {
            for (int j = *returnSize; j > idx; j--) {
                target[j] = target[j - 1];
            }
            target[idx] = nums[i];
            (*returnSize)++;
        }
    }
    return target;
}